File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var value = 0
    for c in s {
        if value < 0 { return false }
        if c == ")" {
            value -= 1    
        } else {
           value += 1   
        }
    }
    
    return value == 0 ? true : false
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let answer = true;
    let num = 0
    for(let i = 0; i < s.length; i++){
        if(num < 0) {
            answer = false;
            break;
        } else if(s[i] === "("){
            num++;
        } else {
            num--;
        }
    }
    if(num !== 0) answer = false;

    return answer;
}